packages/ref-core/src/cmip_ref_core/providers.py

```diff
@@ -86,3 +99,176 @@ def get(self, slug: str) -> Metric:
             The requested metric.
         """
         return self._metrics[slug.lower()]
+
+
+def import_provider(fqn: str) -> MetricsProvider:
+    """
+    Import a provider by name
+
+    Parameters
+    ----------
+    fqn
+        Full package and attribute name of the provider to import
+
+        For example: `cmip_ref_metrics_example.provider` will use the `provider` attribute from the
+        `cmip_ref_metrics_example` package.
+
+        If only a package name is provided, the default attribute name is `provider`.
+
+    Raises
+    ------
+    InvalidProviderException
+        If the provider cannot be imported
+
+        If the provider isn't a valid `MetricsProvider`.
+
+    Returns
+    -------
+    :
+        MetricsProvider instance
+    """
+    if "." in fqn:
+        module, name = fqn.rsplit(".", 1)
+    else:
+        module = fqn
+        name = "provider"
+
+    try:
+        imp = importlib.import_module(module)
+        provider = getattr(imp, name)
+        provider.module = imp  # TODO: fix this hack
+        if not isinstance(provider, MetricsProvider):
+            raise InvalidProviderException(fqn, f"Expected MetricsProvider, got {type(provider)}")
+        return provider
+    except ModuleNotFoundError:
+        logger.error(f"Module '{fqn}' not found")
+        raise InvalidProviderException(fqn, f"Module '{module}' not found")
+    except AttributeError:
+        logger.error(f"Provider '{fqn}' not found")
+        raise InvalidProviderException(fqn, f"Provider '{name}' not found in {module}")
+
+
+class CommandLineMetricsProvider(MetricsProvider):
+    """
+    A metrics provider for metrics that can be run from the command line.
+    """
+
+    @abstractmethod
+    def run(self, cmd: Iterable[str]) -> None:
+        """
+        Return the command to run.
+        """
+
+
+MICROMAMBA_EXE = (
+    "https://github.com/mamba-org/micromamba-releases/releases/latest/download/micromamba-linux-64"
```

Comment:
Should we use the platform-independent installation method instead of assuming an architecture?

```
"${SHELL}" <(curl -L https://micro.mamba.pm/install.sh)
```

Reply:
Done in 7f3fbca